src/adloader.js

```diff
@@ -29,46 +30,60 @@ export function loadExternalScript(url, moduleCode, callback) {
     logError(`${moduleCode} not whitelisted for loading external JavaScript`);
     return;
   }
+  if (!doc) {
+    doc = document; // provide a "valid" key for the WeakMap
+  }
   // only load each asset once
-  if (_requestCache[url]) {
+  const storedCachedObject = getCacheObject(doc, url);
+  if (storedCachedObject) {
     if (callback && typeof callback === 'function') {
-      if (_requestCache[url].loaded) {
+      if (storedCachedObject.loaded) {
         // invokeCallbacks immediately
         callback();
       } else {
         // queue the callback
-        _requestCache[url].callbacks.push(callback);
+        storedCachedObject.callbacks.push(callback);
       }
     }
-    return _requestCache[url].tag;
+    return storedCachedObject.tag;
   }
-  _requestCache[url] = {
+  const cachedDocObj = _requestCache.get(doc) || {};
+  const cacheObject = {
     loaded: false,
     tag: null,
     callbacks: []
   };
+  cachedDocObj[url] = cacheObject;
+  _requestCache.set(doc, cachedDocObj);
+
   if (callback && typeof callback === 'function') {
-    _requestCache[url].callbacks.push(callback);
+    cacheObject.callbacks.push(callback);
   }
 
   logWarn(`module ${moduleCode} is loading external JavaScript`);
   return requestResource(url, function () {
-    _requestCache[url].loaded = true;
+    cacheObject.loaded = true;
     try {
-      for (let i = 0; i < _requestCache[url].callbacks.length; i++) {
-        _requestCache[url].callbacks[i]();
+      for (let i = 0; i < cacheObject.callbacks.length; i++) {
+        cacheObject.callbacks[i]();
       }
     } catch (e) {
       logError('Error executing callback', 'adloader.js:loadExternalScript', e);
     }
-  });
+  }, doc);
 
-  function requestResource(tagSrc, callback) {
-    var jptScript = document.createElement('script');
+  function requestResource(tagSrc, callback, doc) {
+    if (!doc) {
```

Comment:
It depends, ```doc``` being not set defaults to the loaded ```document```, which in theory does also happen down the line in ```insertElement``` and some other methods. But now it's specific from here on. But i do agree it could also be an Argument Exception if there is a need to be specific about this. 
The ```cacheObject``` could theoeratically be gone for what ever reason it's resolved there. This could actually be an error, since the ```tag``` property won't be set, but the callback would flag the object as loaded, which is sort of inconsistent. But not failing there and just loading the script, without failing there is i think not that troublesome.